source/dub/generators/generator.d

```diff
@@ -485,5 +487,12 @@ void runBuildCommands(in string[] commands, in Package pack, in GeneratorSetting
 	env["DUB_PARALLEL_BUILD"]    = settings.parallelBuild? "TRUE" : "";
 
 	env["DUB_RUN_ARGS"] = (cast(string[])settings.runArgs).map!(escapeShellFileName).join(" ");
+	
+	PackagesUsed.add(proj.rootPackage.name);
+	foreach (dep; proj.dependencies)
+		PackagesUsed.add(dep.name);
+	PackagesUsed.store(env);
+	PackagesUsed.load(); // Refresh internal list from actual current environment
```

Comment:
No, I'm deliberately *not* changing the current process's environment. If I did (which I actually did by accident at one point while developing this PR), then dependencies would occasionally get skipped when they shouldn't get skipped.